### PR TITLE
Update Rasplex, PlexBlack, Recalbox and noobs

### PR DESCRIPTION
**Pull request by eloo**
Update to latest versions and use noobs lite for faster build.

### CODE REVIEW

File: package/noobs/noobs.mk [hunk 1]
```diff
@@ -4,9 +4,9 @@
 #
 ################################################################################
 
-NOOBS_VERSION = v2_0_0
-NOOBS_SOURCE = NOOBS_$(NOOBS_VERSION).zip
-NOOBS_SITE = https://downloads.raspberrypi.org/NOOBS/images/NOOBS-2016-10-05
+NOOBS_VERSION = v2_3
```
Comment (LaurentMarchelli) [inline package/noobs/noobs.mk:7]: I guess you have checked it works with the light version right ?

File: package/rasplex/Config.in [hunk 2]
```diff
@@ -17,13 +17,13 @@ choice
     depends on BR2_PACKAGE_RASPLEX
 
 config BR2_PACKAGE_RASPLEX_STABLE
-    bool "1.6.2 - Stable"
+    bool "1.7.1 - Stable"
```
Comment (LaurentMarchelli) [inline package/rasplex/Config.in:20]: Last Stable is 1.6.2 ... 1.7.1 is still pre-release.
Furthermore, your rasplex.mk file does it right it uses the 1.6.2 ... ;-)

File: package/rasplex/rasplex.mk [hunk 3]
```diff
@@ -3,11 +3,28 @@
...
+RASPLEX_SOURCE = RasPlex-$(RASPLEX_VERSION).tar.gz
+RASPLEX_SITE = https://github.com/RasPlex/RasPlex/releases/download/$(RASPLEX_RELEASE)
+RASPLEX_LICENSE = GPL2
+RASPLEX_LICENSE_FILES = COPYING
+
+################################################################################
+#
+# rasplex
+#
+################################################################################
```
Comment (LaurentMarchelli) [inline package/rasplex/rasplex.mk:22]: Why these duplicate lines, there were already defined lines 35 - 43 ?
I guess it is a merge conflict error.

File: package/rasplex/rasplex.mk [hunk 4]
```diff
@@ -135,6 +165,38 @@ define RASPLEX_BUILD_SKIN_CMD
...
+
+			# cp -r $(RASPLEX_PKGDIR)recalplex/addons/$(RASPLEX_SKIN_NAME)/* \
+			# 	$(RASPLEX_TAR_BUILD)/plexdata/.plexht/addons/$(RASPLEX_SKIN_NAME)/;
+		)
+		# Copy customized rasplex skin settings 
+		mkdir -p $(RASPLEX_TAR_BUILD)/plexdata/.plexht/userdata/; \
+		cp -r $(RASPLEX_PKGDIR)recalplex/userdata/$(RASPLEX_SKIN_NAME)/* \
+			$(RASPLEX_TAR_BUILD)/plexdata/.plexht/userdata/; \
+		cp -r $(RASPLEX_PKGDIR)recalplex/slideshow \
+		$(RASPLEX_TAR_BUILD)/plexdata/.recalplex/;
```
Comment (LaurentMarchelli) [inline package/rasplex/rasplex.mk:199]: Why these duplicate lines, there were already defined lines 137 - 168 ?
I guess it is a merge conflict error.

File: package/rasplex/rasplex.mk [hunk 5]
```diff
@@ -90,7 +107,7 @@ ifdef BR2_PACKAGE_RASPLEX_SKIN_PLEX_BLACK_EDITION
 	# URL information stored in /storage/.plexht/userdata/Database/Addons15.db
 	RASPLEX_SKIN_NAME = skin.plex_black_editionHT
 	ifeq ($(BR2_PACKAGE_RASPLEX_PRERELEASE),y)
-		RASPLEX_SKIN_VERSION = 16.11.24
+		RASPLEX_SKIN_VERSION = 17.03.16
```
Comment (LaurentMarchelli) [inline package/rasplex/rasplex.mk:110]: It shouldn't be 18.x.x for the version 18, could you please check.

File: package/rasplex/rasplex.mk [hunk 6]
```diff
@@ -103,8 +120,21 @@ ifdef RASPLEX_SKIN_NAME
...
+	RASPLEX_SKIN_NAME = skin.plex_black_editionHT
+	RASPLEX_SKIN_VERSION = v16.11.24
+	# URL information stored in /storage/.plexht/userdata/Database/Addons15.db
+	RASPLEX_SKIN_SITE = https://addons.openpht.tv/openpht-1.6
+	RASPLEX_SKIN_FILE = $(RASPLEX_SKIN_NAME)-$(RASPLEX_SKIN_VERSION).zip
+	ZIP_NAME = HQlrwa
+	# # Download skin if required by configuration
+	RASPLEX_EXTRA_DOWNLOADS = https://goo.gl/HQlrwa
+endif
+
```
Comment (LaurentMarchelli) [inline package/rasplex/rasplex.mk:134]: Old code, now we are downloading the skin from the official source https://addons.openpht.tv/openpht-1.8, not from https://goo.gl anymore.
I guess it is a merge conflict error.

File: package/recalboxOS/recalboxOS.mk [hunk 7]
```diff
@@ -3,11 +3,16 @@
...
 RECALBOXOS_SOURCE = $(RECALBOXOS_NAME)-$(RECALBOXOS_RELEASE).zip
 RECALBOXOS_SITE = https://github.com/recalbox/recalbox-os/releases/download/$(RECALBOXOS_RELEASE)
 
-ifeq ($(BR2_PACKAGE_RECALBOXOS_UNSTABLE),y)
+ifeq ($(BR2_PACKAGE_RECALBOXOS_STABLE),y)
+	RECALBOXOS_VERSION = 4.0.0
+	RECALBOXOS_EXTRA_DOWNLOADS = \
+		http://archive.recalbox.com/4/$(BR2_ARCH)/stable/last/boot.tar.xz \
+		http://archive.recalbox.com/4/$(BR2_ARCH)/stable/last/root.tar.xz
+else ifeq ($(BR2_PACKAGE_RECALBOXOS_UNSTABLE),y)
```
Comment (LaurentMarchelli) [inline package/recalboxOS/recalboxOS.mk:15]: Extra download is not needed for the stable, as the boot.tar.xz and the root.tar.xz are already included in the zip file.
Following line is enough to replace your block.
ifeq ($(BR2_PACKAGE_RECALBOXOS_UNSTABLE),y)

**Review by LaurentMarchelli — CHANGES_REQUESTED** [review]
Hi Joseph,

Thank for your help, I had to do it, you made my day.
I guess you had some trouble with my master merge, so there are some duplicate codes.

Could you please, check requested corrections.
Thank's,

Laurent

File: package/rasplex/rasplex.mk (hunk 5 above)
Comment (eloo) [reply package/rasplex/rasplex.mk:110]: its a timestamp ;) so the 17 stands for 2017

**Comment by eloo** [comment]
Hi Laurent, yes I was facing some trouble while merging.. 
I will recheck the changes in time.

**Comment by LaurentMarchelli** [comment]
Hi Joseph,
Do you need some help to resync ?
Regards,
Laurent

**Comment by LaurentMarchelli** [comment]
Hi Joseph,

Your changes have been checked and merged with #18, please resynchronize your fork.
Try to avoid working on master branch, as it is your repository default branch, github does not allows the push force.
Using pull -r (rebase) on a development branch is always better than merging.
Thank for your contribution, noobs lite version save bandwidth.

Best regards,
Laurent Marchelli

**Comment by eloo** [comment]
Hi Laurent,

glad to see you've already adapted my changes.
Sorry that i cant fix my PR but I was busy last weeks. 
Okay yes will try to work on feature branche in future ;)
Is there a special reason why you are using noobs lite 2.1 instead of 2.3?

Best regards,
Joseph

**Comment by LaurentMarchelli** [comment]
Hi Joseph,

No problem, when I didn't see any answer I guessed you were busy and as I could handle it ... It's the way a team works, isn't it ? ;-)

I downgraded noobs to version 2.1 because there is a bug since version 2.2, Noobs does not take care of autoboot.txt, so you get the noobs boot menu each time you try to switch, check my commit comment :
Noobs_lite version 2_3 does not work with autoboot.txt raspberrypi/noobs#390

In another hand, because I changed the url of the .buildroot submodule, you may have trouble to resync your fork, so let me know if you need any help to resync and clean your master branch.

Best regards,
Laurent Marchelli

**Comment by eloo** [comment]
Okay.. yeah may i will remove my fork and re-fork next time i will contribute.. i guess this is the best solution to work on a clean tree ;)

**Comment by LaurentMarchelli** [comment]
As u like, : 
- Creating a new fork is easier and faster.
- Resynchronizing your fork may be harder, but may learn more.
Let me know. ;-)

Best Regards,
Laurent Marchelli

**Pull request closed without merging**